examples/scanner/scanner.ino

```diff
@@ -8,143 +8,113 @@
+ * While any 2.4GHz device can trigger this carrier wave detection, a second NRF24L01+ device can be configured to output a constant 
+ * carrier wave by sending the character "g" over the serial monitor. Output power and channel can be adjusted in the #defines below.
  *
- * Run this sketch on two devices. On one device, start CCW output by sending a 'g'
- * character over Serial. The other device scanning should detect the output of the sending
- * device on the given channel. Adjust channel and output power of CCW below.
- *
- * Inspired by cpixip.
- * See http://arduino.cc/forum/index.php/topic,54795.0.html
- */
+ * Originally inspired by cpixip,
+ * See https://forum.arduino.cc/t/poor-mans-2-4-ghz-scanner/54846/39
+*/
 
 #include "RF24.h"
-#include "printf.h"
-
-//
-// Hardware configuration
-//
 
-// Set up nRF24L01 radio on SPI bus plus pins 7 & 8
+#define CARRIER_POWER RF24_PA_LOW  //can be RF24_PA_MIN, RF24_PA_LOW, RF24_PA_HIGH, RF24_PA_MAX
+#define CARRIER_CHANNEL 40         //0-125
 
-RF24 radio(7, 8);
+// Set up nRF24L01 radio on SPI bus plus pins 9 & 10 for CE and CSN
+RF24 radio(5, 6);
 
-//
 // Channel info
-//
-
 const uint8_t num_channels = 126;
-uint8_t values[num_channels];
+uint8_t values[num_channels];  //number of times a value is detected on each channel during the scan
 
-//
-// Setup
-//
+const int num_reps = 100;  //number of times to scan through channels
+bool constCarrierMode = 0;
 
 void setup(void) {
-  //
-  // Print preamble
-  //
 
+  // Print preamble
   Serial.begin(115200);
-  printf_begin();
-  Serial.println(F("\n\rRF24/examples/scanner/"));
+  while (!Serial)
+    ;
+  Serial.println(F("rRF24/examples/scanner/"));
 
-  //
   // Setup and configure rf radio
-  //
+  if(!radio.begin())
+  {
+    Serial.println("Can't connect to radio");
+    while(1);
+  }
 
-  radio.begin();
   radio.setAutoAck(false);
-
-  // Get into standby mode
-  radio.startListening();
   radio.stopListening();
   radio.printDetails();
 
-  //delay(1000);
   // Print out header, high then low digit
-  int i = 0;
-  while (i < num_channels) {
+  for(uint8_t i = 0; i < num_channels; i++) {
     Serial.print(i >> 4, HEX);
-    ++i;
   }
   Serial.println();
-  i = 0;
-  while (i < num_channels) {
+
+  for(uint8_t i = 0; i < num_channels; i++) {
     Serial.print(i & 0xf, HEX);
-    ++i;
   }
   Serial.println();
-  //delay(1000);
 }
 
-//
 // Loop
-//
-
-const int num_reps = 100;
-bool constCarrierMode = 0;
-
 void loop(void) {
-  /****************************************/
-  // Send g over Serial to begin CCW output
-  // Configure the channel and power level below
+  //check for g to begin carrier wave output
   if (Serial.available()) {
     char c = Serial.read();
     if (c == 'g') {
       constCarrierMode = 1;
       radio.stopListening();
       delay(2);
-      Serial.println("Starting Carrier Out");
-      radio.startConstCarrier(RF24_PA_LOW, 40);
+      Serial.println("Starting Carrier Out, send \"e\" to return to scanning");
```

Comment:
We prefer to use `F()` macro function to keep static strings in the MCU flash
```suggestion
      Serial.println(F("Starting Carrier Out, send 'e' to return to scanning"));
```
And it would make more sense to use single quotes since it is a single byte being sent over Serial monitor.